x-pack/plugins/lens/public/indexpattern_plugin/indexpattern.tsx

```diff
@@ -55,16 +56,25 @@ export const indexPatternDatasource: Datasource<
   IndexPatternPersistedState
 > = {
   async initialize(state?: IndexPatternPersistedState) {
+    const indexPatternObjects = await getIndexPatterns();
+    const indexPatterns: { [id: string]: IndexPattern } = {};
```

Comment:
nit: Not necessary, but this can also be written as `Record<string, IndexPattern>`. By doing so it's you don't have to give a name to the props which will never be used again